### ISSUE TITLE
Syntax highlighting confused with division of postincrement

### ISSUE THREAD

**Issue by choroba** [jrockway/cperl-mode#45]
`/` is highlighted as a start of a regex instead of division after `++` or `--`. The rest of the code is highlighted wrongly.

![screenshot from 2017-09-13 11-55-30](https://user-images.githubusercontent.com/886196/30371304-7c1299cc-987a-11e7-80b1-91a4308aa2de.png)

**Comment by renormalist** [comment]
Sigh, that magic slash.

I just know I once debugged a similar problem with variables named $y and slashes in a non-regex context but to cperl looking like y/// (the alternative syntax for tr///) -- and lost the battle.

Patches welcome. :-)

**Comment by choroba** [comment]
I was able to identify the place that changes the behaviour, but I can't find a way to fix it. For example, the following fixes division, but breaks matching:

![screenshot from 2018-10-12 22-56-57](https://user-images.githubusercontent.com/886196/46893752-5441eb80-ce72-11e8-828a-f41f50167371.png)

```diff
modified   cperl-mode.el
@@ -4126,8 +4126,8 @@ the sections using `cperl-pod-head-face', `cperl-pod-face',
 				(or (memq (preceding-char)
 					  (append (if (memq c '(?\? ?\<))
 						      ;; $a++ ? 1 : 2
-						      "~{(=|&*!,;:["
-						    "~{(=|&+-*!,;:[") nil))
+						      "~{(=|&+-*!,;:["
+						    "~{(=|&*!,;:[") nil))
 				    (and (eq (preceding-char) ?\})
 					 (cperl-after-block-p (point-min)))
 				    (and (eq (char-syntax (preceding-char)) ?w)
```

**Comment by choroba** [comment]
Fixed by #55
